src/components/input/clear/index.tsx

```diff
@@ -43,7 +43,8 @@ export class SmoothlyInputClear {
 		})
 	}
 	@Listen("click")
-	clickHandler() {
+	clickHandler(event: UIEvent) {
+		event.stopPropagation()
```

Comment:
instead of stopping the propagation. check if the event came from this div:
```tsx
<div class="icons">
	<slot name="end" />
	<smoothly-icon size="tiny" name={this.open ? "caret-down-outline" : "caret-forward-outline"} />
</div>
```
and didn't come from the icon. If those conditions are met don't toggle / close the menu on the click event.

If i as a user of the select component want to know if a user clicked on the select component in any way i will not know about it if the event is stopped.

Reply:
i have changed it to see if an inside button was clicked, that div contains the symbol of opening and closing menu so i wanted to keep it clickable!